frontend/src/hooks/useAppContext.tsx

```diff
@@ -0,0 +1,61 @@
+import { useContext } from 'react'
+import { AppContext } from '../contexts/app-context/AppContext'
+
+const useAppContext = () => {
+    const {
+        shoppingCartItems,
+        setShoppingCartItems,
+        userSignedIn,
+        setUserSignedIn,
+        shippingAddressData,
+        setShippingAddressData,
+        progressStep,
+        setProgressStep,
+        setPaymentMethod,
+        paymentMethod,
+        orderData,
+        setOrderData,
+        userOptionsOpen,
+        setUserOptionsOpen,
+        userOptionsOpenMobile,
+        setUserOptionsOpenMobile,
+        isMenuOpen,
+        isMobileMenuOpen,
+        handleMobileMenuClose,
+        handleMenuClose,
+        handleMobileMenuOpen,
+        handleNavigation,
+        getMenuClickHandler,
+        handleProfileMenuOpen,
+        handleSignOut
+    } = useContext(AppContext)
```

Comment:
this context is doing way tooooo many things, why not have a couple of contexts instead?
examples:
`usePaymentContext`
`useCartContext`

Reply:
You sure The convention for naming context starts with  -use-  ?